src/app/trips/shared/trip.service.ts

```diff
@@ -1,33 +1,47 @@
-import { LocationStrategy } from "@angular/common";
 import { Injectable } from "@angular/core";
-import { ILocation, ITrip } from "./trip.model";
+import { Observable, Subject } from "rxjs";
+import { ILocation } from "./location.model";
+import { ITrip } from "./trip.model";
 
 @Injectable()
 export class TripService{
 
-  getAllTrips() {
-    return TRIPS
+  getAllTrips():Observable<ITrip[]> {
+    let subject = new Subject<ITrip[]>()
+    setTimeout(() => {subject.next(TRIPS); subject.complete();}, 100)
```

Comment:
We should talk through what the goal is here. Typically `setTimeout` to work with observables is a hack because... well observables can be hard. Let's remove the `setTimeout` and talk through any problems you might have once we understand the goal.

Reply:
if the goal for now is to just return the list of trips you can do `of(TRIPS)` and that will be an observable until you have an API